param/assembler/assembler.go

```diff
@@ -0,0 +1,286 @@
+// Package assembler takes a collection of FormPair tuples and uses them to
+// construct more complex data types using "Rack-style" conventions for arrays
+// and maps with a few small Stripe-specific tweaks.
+//
+// When processing a request, data should first be parsed from the query, form,
+// or multipart form, handed off to this package for assembly, then passed on
+// the coercer to coerce string to other expected types.
+package assembler
```

Comment:
The corresponding function in Rack is called `Rack::Utils.parse_nested_query`. I wonder if we could choose a name that suggests the connection?

Reply:
Yeah. I'd kind of like to have a name that doesn't include "parse" though, because we have other modules that do the real string parsing and are more deserved of the name.

I don't mind "assembler" that much. I'm trying to think of some alternatives like "typebuilder", but don't like how that can so easily be confused with types from the programming language. Maybe "nestedformbuilder"? That's starting to get a little wordy though.

Reply:
Maybe `nestedqueryassembler`? Or even just `nestedquery`? I'll leave it up to you.

Reply:
Thanks! Alright, I ended up going with `nestedtypeassembler`.